v2/cmd/controller/app/setup.go

```diff
@@ -74,6 +76,11 @@ func SetupPreUpgradeCheck(ctx context.Context) error {
 			continue
 		}
 
+		if !scheme.Recognizes(crd.GroupVersionKind()) {
```

Comment:
Maybe combine this with the check above? 

```
if !crdRegexp.MatchString(crd.Name) || !scheme.Recognizes(crd.GroupVersionKind()){
  continue
}
```

Reply:
I don't think that adds any clarity.